linode/manager
# refactor: [M3-7229] - Migrate UserDefinedSelect from a TextField Select to an Autocomplete

## Description 📝

- Relates to (this is the last PR like this)
  - https://github.com/linode/manager/pull/9752
  - https://github.com/linode/manager/pull/9754
- In order to fix up our TextField component, we need to not allow it to be used as a select component. This PR replaces the TextField Select in the NodeBalancer config node component with a modern alternative like Autocomplete

## Preview 📷

| Before  | After   |
| ------- | ------- |
| ![Screenshot 2023-10-04 at 1 33 01 PM](https://github.com/linode/manager/assets/115251059/53238609-9555-4e3e-865a-2e50eeec704d) | ![Screenshot 2023-10-04 at 1 32 21 PM](https://github.com/linode/manager/assets/115251059/d26aeb3e-9abd-4499-ac2b-eb1f878ff82d) |

## How to test 🧪
- Test the functionality of Marketplace App Select Components
- This is testable on the Minecraft marketplace app
- Verify `Player Idle Timeout` is selectable and shows the selected value (`playeridletimeout`) when the Linode `POST` happens
